### PR TITLE
test: permissions

### PR DESCRIPTION
**Pull request by cajubelt**

### CODE REVIEW

File: contracts/mocks/MockRouter.sol [hunk 1]
```diff
@@ -5,9 +5,6 @@ import "./MockLPToken.sol";
 
 contract MockRouter is IUniswapV2Router01 {
     MockLPToken lpToken;
-    constructor(address _lpToken) {
```
Comment (cajubelt) [inline contracts/mocks/MockRouter.sol:8]: it doesn't make sense for a router to have a single lpToken member, but we need to know what LP token to mint for `addLiquidity`. so I removed it from the constructor and added a setter instead, which seems slightly less bad

**Comment by cajubelt** [comment]
should go after #35

File: test/farm-bot-test.ts [hunk 2]
```diff
@@ -1,57 +1,82 @@
...
 
 const {ethers} = require("hardhat")
 
 
 describe('Farm bot tests', () => {
-  it('Able to deploy farm bot to local test chain', async () => {
-    const [owner, reserve] = await ethers.getSigners()
-    const revoFeesFactory = await ethers.getContractFactory('MockRevoFees')
-    const feeContract: MockRevoFees = await revoFeesFactory.deploy()
+  let deployer: SignerWithAddress, reserve: SignerWithAddress, compounder: SignerWithAddress, investor: SignerWithAddress,
```
Comment (jophish) [inline test/farm-bot-test.ts:21]: that's a lot of state! :sweat_smile:

**Review by jophish — APPROVED** [review]
This looks fantastic; it looks like we have a great framework now to (somewhat) easily add new tests :slightly_smiling_face:

File: test/farm-bot-test.ts (hunk 2 above)
Comment (cajubelt) [reply test/farm-bot-test.ts:21]: it really is

**Pull request merged**